src/services/meta.ts

```diff
@@ -22,6 +22,10 @@ export class MetaService extends Service {
     return fetch(this.buildUrl(`vaults/${CHAIN_ID_KEY}/all`)).then(res => res.json());
   }
 
+  async findOne(address: Address): Promise<VaultMetadataOverrides> {
```

Comment:
@karelianpie What do you think if we use the same `vaults()` method and add an additional optional `addresses?: Address` parameter? Seems like we that pattern is used through out all sdk instead of adding a new method.

If possible, would do the same for `tokens()` in same PR.

Reply:
I think that doesn't work for this case because the return type of that function is an array of `VaultMetadataOverrides`:
```typescript
async vaults(): Promise<VaultMetadataOverrides[]> {}
```
Whereas we just literally want one, I'm happy to refactor and rename it `vault` instead:
```typescript
async vault(address: Address): Promise<VaultMetadataOverrides> {}
```

Reply:
I like the idea, but not for when we intend to explicitly get a single resource.

I've refactored with https://github.com/yearn/yearn-sdk/pull/263/commits/1dcd5067269f6cbdd405c33b5e96f675fa8d63ed

So now, both `tokens` and `vaults` take an optional array of addresses and if given, they will fetch those addresses individually and merge into a metadata array.

```typescript
async tokens(addresses?: Address[]): Promise<TokenMetadata[]> {}

async vaults(addresses?: Address[]): Promise<VaultMetadataOverrides[]> {}
```

However, I'd like to keep the single functions because they are much easier to manage in terms of return type and their intention is much clearer:

```typescript
async token(address: Address): Promise<TokenMetadata> {}

async vault(address: Address): Promise<VaultMetadataOverrides> {}
```

It feels wrong to knowing that I want a single resource, to pass it inside an array and then having an array returned to me where I know there's only one item. In usage, the difference is

```typescript
// with a function for a single address
this.yearn.services.meta.vault(address);

// giving a single address to a function that takes an array of addresses
(await this.yearn.services.meta.vaults([address]))[0];
```

Reply:
Added a test for this service with https://github.com/yearn/yearn-sdk/pull/263#:~:text=test%3A%20Add%20test%20for%20MetaService